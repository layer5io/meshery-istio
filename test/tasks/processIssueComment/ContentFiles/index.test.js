const fs = require('fs')
const path = require('path')

const ContentFiles = require('../../../../src/tasks/processIssueComment/ContentFiles')

describe('ContentFiles', () => {
    const mockRepository = {}
    const mockTestFileContent = fs.readFileSync(
        path.join(__dirname, 'test-readme-file.md'),
        'utf8',
    )
    const mockTestFileContentNoTable = fs.readFileSync(
        path.join(__dirname, 'test-readme-file-no-table.md'),
        'utf8',
    )

    test(`Add's new contributor`, () => {
        const contentFiles = new ContentFiles({
            repository: mockRepository,
        })
        contentFiles.contentFilesByPath = {
            'README.md': {
                content: mockTestFileContent,
            },
        }

        const mockOptionsConfig = {
            get: function() {
                return {
                    contributorsPerLine: 7,
                    contributors: [
                        {
                            login: 'jakebolam',
                            name: 'Jake Bolam',
                            avatar_url:
                                'https://avatars2.githubusercontent.com/u/3534236?v=4',
                            profile: 'https://jakebolam.com',
                            contributions: ['code', 'ideas', 'infra', 'test'],
                        },
                        {
                            login: 'tbenning',
                            name: 'tbenning',
                            avatar_url:
                                'https://avatars2.githubusercontent.com/u/7265547?v=4',
                            profile: 'https://github.com/tbenning',
                            contributions: ['design'],
                        },
                    ],
                }
            },
        }
        contentFiles.generate(mockOptionsConfig)

        expect(contentFiles.get()['README.md'].content).toMatchSnapshot()
    })

    test(`Init content`, () => {
        const contentFiles = new ContentFiles({
            repository: mockRepository,
        })
        const mockSha = 'mock sha right here yo'
        contentFiles.contentFilesByPath = {
            'README.md': {
                content: mockTestFileContentNoTable,
                sha: mockSha,
            },
        }

        const mockOptionsConfig = {
            get: function() {
                return {
                    contributorsPerLine: 7,
                }
            },
        }
        contentFiles.init(mockOptionsConfig)

        const readme = contentFiles.get()['README.md']
        expect(readme.content).toMatchSnapshot()
        expect(readme.originalSha).toEqual(mockSha)

        const mockOptionsConfig2 = {
            get: function() {
                return {
                    contributorsPerLine: 7,
                    contributors: [
                        {
                            login: 'jakebolam',
                            name: 'Jake Bolam',
                            avatar_url:
                                'https://avatars2.githubusercontent.com/u/3534236?v=4',
                            profile: 'https://jakebolam.com',
                            contributions: ['code', 'ideas', 'infra', 'test'],
                        },
                        {
                            login: 'tbenning',
                            name: 'tbenning',
                            avatar_url:
                                'https://avatars2.githubusercontent.com/u/7265547?v=4',
                            profile: 'https://github.com/tbenning',
                            contributions: ['design'],
                        },
                    ],
                }
            },
        }
        contentFiles.generate(mockOptionsConfig2)

        const readme2 = contentFiles.get()['README.md']
        expect(readme2.content).toMatchSnapshot()
        expect(readme2.originalSha).toEqual(mockSha)
    })
})
