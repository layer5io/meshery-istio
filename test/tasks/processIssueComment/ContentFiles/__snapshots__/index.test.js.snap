// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentFiles Add's new contributor 1`] = `
"# TestContentFIles

[![All Contributors](https://img.shields.io/badge/all_contributors-2-orange.svg?style=flat-square)](#contributors-)

A test for content files generation

### Content
1. Stuff here
2. Yeah boy


## Contributors

Thanks goes to these wonderful people ([emoji key](https://github.com/all-contributors/all-contributors#emoji-key)):

<!-- ALL-CONTRIBUTORS-LIST:START - Do not remove or modify this section -->
<!-- prettier-ignore-start -->
<!-- markdownlint-disable -->
<table>
  <tr>
    <td align=\\"center\\"><a href=\\"https://jakebolam.com\\"><img src=\\"https://avatars2.githubusercontent.com/u/3534236?v=4\\" width=\\"100px;\\" alt=\\"Jake Bolam\\"/><br /><sub><b>Jake Bolam</b></sub></a><br /><a href=\\"#code-jakebolam\\" title=\\"Code\\">üíª</a> <a href=\\"#ideas-jakebolam\\" title=\\"Ideas, Planning, & Feedback\\">ü§î</a> <a href=\\"#infra-jakebolam\\" title=\\"Infrastructure (Hosting, Build-Tools, etc)\\">üöá</a> <a href=\\"#test-jakebolam\\" title=\\"Tests\\">‚ö†Ô∏è</a></td>
    <td align=\\"center\\"><a href=\\"https://github.com/tbenning\\"><img src=\\"https://avatars2.githubusercontent.com/u/7265547?v=4\\" width=\\"100px;\\" alt=\\"tbenning\\"/><br /><sub><b>tbenning</b></sub></a><br /><a href=\\"#design-tbenning\\" title=\\"Design\\">üé®</a></td>
  </tr>
</table>

<!-- markdownlint-enable -->
<!-- prettier-ignore-end -->
<!-- ALL-CONTRIBUTORS-LIST:END -->

This project follows the [all-contributors](https://github.com/all-contributors/all-contributors) specification. Contributions of any kind welcome

## LICENSE

[MIT](LICENSE)
"
`;

exports[`ContentFiles Init content 1`] = `
"# TestContentFIles
[![All Contributors](https://img.shields.io/badge/all_contributors-0-orange.svg?style=flat-square)](#contributors-)

A test for content files generation

### Content
1. Stuff here
2. Yeah boy

## LICENSE

[MIT](LICENSE)

## Contributors ‚ú®

Thanks goes to these wonderful people ([emoji key](https://allcontributors.org/docs/en/emoji-key)):

<!-- ALL-CONTRIBUTORS-LIST:START - Do not remove or modify this section -->
<!-- prettier-ignore-start -->
<!-- markdownlint-disable -->
<!-- markdownlint-enable -->
<!-- prettier-ignore-end -->
<!-- ALL-CONTRIBUTORS-LIST:END -->

This project follows the [all-contributors](https://github.com/all-contributors/all-contributors) specification. Contributions of any kind welcome!"
`;

exports[`ContentFiles Init content 2`] = `
"# TestContentFIles
[![All Contributors](https://img.shields.io/badge/all_contributors-2-orange.svg?style=flat-square)](#contributors-)

A test for content files generation

### Content
1. Stuff here
2. Yeah boy

## LICENSE

[MIT](LICENSE)

## Contributors ‚ú®

Thanks goes to these wonderful people ([emoji key](https://allcontributors.org/docs/en/emoji-key)):

<!-- ALL-CONTRIBUTORS-LIST:START - Do not remove or modify this section -->
<!-- prettier-ignore-start -->
<!-- markdownlint-disable -->
<table>
  <tr>
    <td align=\\"center\\"><a href=\\"https://jakebolam.com\\"><img src=\\"https://avatars2.githubusercontent.com/u/3534236?v=4\\" width=\\"100px;\\" alt=\\"Jake Bolam\\"/><br /><sub><b>Jake Bolam</b></sub></a><br /><a href=\\"#code-jakebolam\\" title=\\"Code\\">üíª</a> <a href=\\"#ideas-jakebolam\\" title=\\"Ideas, Planning, & Feedback\\">ü§î</a> <a href=\\"#infra-jakebolam\\" title=\\"Infrastructure (Hosting, Build-Tools, etc)\\">üöá</a> <a href=\\"#test-jakebolam\\" title=\\"Tests\\">‚ö†Ô∏è</a></td>
    <td align=\\"center\\"><a href=\\"https://github.com/tbenning\\"><img src=\\"https://avatars2.githubusercontent.com/u/7265547?v=4\\" width=\\"100px;\\" alt=\\"tbenning\\"/><br /><sub><b>tbenning</b></sub></a><br /><a href=\\"#design-tbenning\\" title=\\"Design\\">üé®</a></td>
  </tr>
</table>

<!-- markdownlint-enable -->
<!-- prettier-ignore-end -->
<!-- ALL-CONTRIBUTORS-LIST:END -->

This project follows the [all-contributors](https://github.com/all-contributors/all-contributors) specification. Contributions of any kind welcome!"
`;
